lib/wallet/wallet.js

```diff
@@ -3400,6 +3400,19 @@ class Wallet extends EventEmitter {
     if (ancestors.size + 1 > this.maxAncestors)
       throw new Error('TX exceeds maximum unconfirmed ancestors.');
 
+    for (const output of tx.outputs) {
+      if (output.isDust())
+        throw new Error('Output is dust.');
+
+      if (output.value > 0) {
+        if (!output.address)
+          throw new Error('Cannot send to unknown address.');
+
+        if (output.address.isNull())
```

Comment:
What if somebody wants to provably burn some HNS? This would prevent somebody from using the API  to do so. I don't see a consensus or policy rule that prevents such an output from existing.

Reply:
I think they can use a nulldata address then. This check I think is to catch user error when an address isn't specified in the output and the null value remains:

https://github.com/handshake-org/hsd/blob/940f463259e760f2c60d0b5a53a5bdab1dd83213/lib/primitives/address.js#L99-L112

Reply:
Ok, looking at the implementation of `isNull`, it doesn't check the address.version so this would still let somebody provably burn since its not checking `Address.isNulldata`

Reply:
The same block of code is here: https://github.com/handshake-org/hsd/blob/89d3ab4c81969e6e14306d567c8f1411ad36c254/lib/wallet/wallet.js#L3288

Reply:
Right - if they used nulldata (which is just address version 31) they just couldn't ALSO use a `ZERO_HASH` as the data part.